packages/cdk8s-cli/README.md

```diff
@@ -32,24 +34,148 @@ $ cd my-fun-little-project
 
 To generate constructs for all Kubernetes API objects of a certain version, use
 the `import` subcommand:
 
 ```shell
-$ cdk8s import SPEC
+$ cdk8s import k8s
 ```
 
-Currently, the only supported spec is `"k8s"` which represents the core
-Kubernetes API. The import command will create a file under `imports/k8s.ts`
-with constructs for each API object in the spec.
+By default, the import command will create a file under `imports/k8s.ts`
+with constructs for each API object in the k8s spec.
 
-The following example will import the latest version of the Kubernetes API
-objects:
+The following example will import a specific version of the Kubernetes API objects:
 
 ```shell
-$ cdk8s import k8s
+$ cdk8s import k8s@1.16.0
+```
+
+Alternatively, you can specify your `k8s` import in the `cdk8s.yaml` config file:
+
+```yaml
+imports:
+  - k8s@1.17.0
+```
+
+#### Selecting `apiVersion`s for specific objects
+
+By default, the command `cdk8s import k8s` will use the latest stable version of k8s objects.
+
+For example, when using k8s 1.17, if you want to use `DeploymentSpec`, you would be importing it from [Deployment v1](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.17/#deploymentspec-v1-apps) by default. 
+
+In the default `v1`, the `revisionHistoryLimit`, which is the number of old ReplicaSets to retain, is set to 10 by default. You would have to explicitly set this to a higher number if you want to retain more.
+
+Let's say that you want the behavior of [extensions/v1beta1](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.17/#deploymentspec-v1beta1-extensions), which defaults to retaining _all_ old ReplicaSets.
+
+To do this, you could add the `--include` flag to your `cdk8s import`. Here is how that would look:
+
+`cdk8s` CLI
+
+```code
+cdk8s import k8s --language typescript --include io.k8s.api.extensions.v1beta1.Deployment
+```
+
+`cdk8s.yaml` Config
+
+This is not yet supported in `cdk8s.yaml` config.
+
+Here we are explicitly choosing to use the [Deployment extensions/v1beta1](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.17/#deployment-v1beta1-extensions) version, which includes the new default behavior that we want from `DeploymentSpec`
+
+### Import Custom Resource Definitions (CRDs)
+
+You can import CRDs from local files (`my_crd.yml`) or from a remote host (`https://raw.githubusercontent.com/jenkinsci/kubernetes-operator/master/deploy/crds/jenkins.io_jenkins_crd.yaml`)
+
+Just like other imports, this is how you do that:
+
+`cdk8s` CLI
+
+```code
+cdk8s import my_crd.yaml
+```
+
+`cdk8s.yaml` config
+
+```yaml
+imports:
+  - my_crd.yaml
+```
+
+By default, your imported CRD will be named by its `spec.names.kind`. If your CRD file contains multiple CRDs, this will create multiple imports, each named by its respective `spec.names.kind`.
+
+For example, if `my_crd.yaml` contained two kinds, `cluster` and `autoscaler`, you would have two imports:
+
+```
+imports/
+  cluster.ts
```

Comment:
Again, replace this with an examples on how to consume this import from each supported language.